blackbox/handlers/storage/google_drive.py

```diff
@@ -0,0 +1,252 @@
+        """
+        Search for an existing folder by path within a specific parent.
+
+        Args
+            folder_path: The path to the folder. It does not matter if there is a
+                         leading or trailing slash.
+            parent_id: The ID of the parent folder. Default is "root".
+
+        Return
+            The ID of the found folder, or None if no folder was found.
+        """
+
+        folder_names = folder_path.split("/")
+        last_folder_found = None
+        for folder_name in folder_names:
+            # Search for folders with this name, of type folder, that are not in the
+            # trash, and who have the provided parent folder ID as an ancestor
+            query = (
+                f"name='{folder_name}' "
+                f"and mimeType='application/vnd.google-apps.folder' "
+                f"and trashed=false and '{parent_id}' in parents"
+            )
+            response = self.client.files().list(
+                q=query,
+                spaces="drive",
+                fields="files(id, name)",
+            ).execute()
+            folders = response.get("files", [])
+            if folders:
+                parent_id = folders[0]["id"]
+                last_folder_found = folders[0]
+            else:
+                break
+
+        return last_folder_found
+
+    def _create_folder(self, folder_path: str, parent_id: str = "root") -> str:
+        """
+        Create a folder in Google Drive.
+
+        Important note: If a folder with this name already exists, another will be
+        created. The original will not be overwritten.
+
+        Args
+            folder_path: The path to the folder. It does not matter if there is a
+                         leading or trailing slash.
+            parent_id: The ID of the parent folder. Default is "root".
+
+        Return
+            The ID of the created folder.
+        """
+
+        folder_names = folder_path.split("/")
+        for folder_name in folder_names:
+            metadata = {
+                "name": folder_name,
+                "mimeType": "application/vnd.google-apps.folder",
+                "parents": [parent_id],
+            }
+            folder = self.client.files().create(
+                body=metadata,
+                fields="id",
+            ).execute()
+            parent_id = folder.get("id")
+        return parent_id
+    
+    def _get_and_ensure_deepest_folder_id(self, path: str) -> str:
+        """
+        Get the ID of the deepest folder in the provided path.
+
+        Any folders in the path that do not already exist will be created.
+
+        Args
+            path: The folder path, excluding a file. It does not matter if there is a
+                  leading or trailing slash.
+        
+        Return
+            The ID of the deepest folder.
+        """
+
+        deepest_folder = path.split("/")[-1]  # Get the last part of the path
```

Comment:
won't this be `' '` if there's a trailing slash?

Seems like if I'm feeding this method the input `/my/awesome/path/`, a more intuitive result would be for it to capture `path`